nstar/src/main.rs

```diff
@@ -12,141 +12,250 @@
+pub enum Subcommand {
+    /// List available containers
+    #[structopt(alias = "ls", alias = "list")]
+    Containers,
+    /// List configured repositories
+    #[structopt(alias = "repos")]
+    Repositories,
+    /// Mount a container
+    Mount {
+        /// Container name
+        name: String,
+        /// Container version
+        version: Version,
+    },
+    /// Umount a container
+    Umount {
+        /// Container name
+        name: String,
+        /// Container version
+        version: Version,
+    },
+    /// Start a container
+    Start {
+        /// Container name
+        name: String,
+        /// Container version
+        version: Version,
+    },
+    /// Stop a container
+    Stop {
+        /// Container name
+        name: String,
+        /// Container version
+        version: Version,
+        /// Timeout
+        #[structopt(default_value = "5")]
+        timeout: u64,
+    },
+    /// Install a npk
+    Install {
+        /// Path to the .npk file
+        npk: PathBuf,
+        /// Target repository
+        repositoriy: String,
+    },
+    /// Uninstall a container
+    Uninstall {
+        /// Container name
+        name: String,
+        /// Container version
+        version: Version,
+    },
+    /// Shutdown Northstar
+    Shutdown,
+    /// Notifications
+    Notifications {
+        /// Exit after n notifications
+        #[structopt(short, long)]
+        number: Option<usize>,
+    },
+    /// Shell completion script generation
+    Completion {
+        /// Output diretory where to generate completions into
+        #[structopt(short, long)]
+        output: PathBuf,
+        /// Generate completions for shell type
+        #[structopt(short, long)]
+        shell: clap::Shell,
+    },
 }
 
-/// Tries to connect to northstar indefinitely
-async fn try_connect<W: Write>(mut output: W, url: &url::Url) -> Result<api::client::Client> {
-    writeln!(output, "Connecting to {}", url)?;
-    loop {
-        match api::client::Client::new(url, Some(100))
-            .await
-            .context(format!("Failed to connect to {}", url))
-        {
-            Err(e) => writeln!(output, "{}", e.to_string().red())?,
-            client => break client,
-        }
-        time::sleep(time::Duration::from_secs(1)).await;
-    }
+/// CLI
+#[derive(StructOpt)]
+pub struct Opt {
+    /// Northstar address
+    #[structopt(short, long, default_value = DEFAULT_HOST)]
+    pub host: url::Url,
+    /// Output json
+    #[structopt(short, long)]
+    pub json: bool,
+    /// Connect timeout in seconds
+    #[structopt(short, long, default_value = "10", parse(try_from_str = parse_duration))]
+    pub timeout: time::Duration,
+    /// Command
+    #[structopt(subcommand)]
+    pub command: Subcommand,
+}
+
+/// Parse a str containing a u64 into a `std::time::Duration` and take the value
+/// value as seconds
+fn parse_duration(src: &str) -> Result<time::Duration, anyhow::Error> {
```

Comment:
nit: maybe consider naming it to something like `parse_secs` just to signal the unit casting in the signature.

Reply:
Done.